x4.md

```diff
@@ -54,325 +49,93 @@ The protocol should be a reflection of our principles.
+# Censorship resistance
 
-In order to receive historic messages from a filter, p2p messages MUST be allowed when creating the filter. Receiving p2p messages is implemented in [geth's Whisper V6 implementation](https://github.com/ethereum/go-ethereum/blob/v1.8.23/whisper/whisperv6/whisper.go#L739-L751).
+TODO
 
-## Anonymity concerns
+# Availability/Robustness
 
-In order to use a mail server, a given node needs to connect to it directly, i.e. add the mail server as its peer and mark it as trusted. This means that the mail server is able to send direct p2p messages to the node instead of broadcasting them. Effectively, it knows which topics the node is interested in, when it is online as well as many metadata like IP address.
+TODO
 
-# Whisper V6 extensions (or Status Whisper Node)
+# Continuance
 
-Protocol's target is to be compliant with [the Whisper V6 Specification](https://github.com/ethereum/go-ethereum/wiki/Whisper). It should not matter which implementation is used as long as the implementation follow the Whisper V6 Specification.
+TODO
 
-However, we added a few extensions, message codes and RPC methods to the Whisper V6 service in order to provide better user experience or due to efficiency requirements.
+# Existing implementations
 
-All described addons are implemented in [status-im/whisper fork](https://github.com/status-im/whisper).
+TODO
 
-## New RPC methods
+# Known issues
 
-TODO: provide a list of RPC methods from `shhext` API which are relevant to this spec.
+## Dependencies between layers
 
+As it was mentioned above, there are cases when two layers dependen on one another:
+* encrypting messages is done by the privacy layer instead of security layer
+* the privacy layer needs to know some details of the payload to properly route it.
 
-## Raw SoK TODO leftovers
+In the next version of the protocol, we will try to remove these dependencies so that all layers are orthogonal and if necessary there is a clean interface between one and another.
 
-## Misc notes
+# TODO
 
-### Factors not captured
-- Censorship-resistance, DPI etc
-- Continuance of protocol
-- Availability/Robustness e.g. FW port blocking
-- p2p no central source of control
-- trust minimized
-- Open machine readable specification
-- multiple implementaions
-- group chat performance
-- mobile specific / churnable
+## Review each section
 
-### TODO: Later, for each property have
+Each section, if possible, should follow this structure:
 - Current protocl
 - How we use it
 - Future considerations
 - Desired by Status
-- The ethereum txn viewpoint
\ No newline at end of file
+- The ethereum txn viewpoint
```

Comment:
@oskarth what do you mean by this?

Reply:
You can look at Ethereum transactions as a form of messaging, with certain guarantees etc

Currently this relies on Infura, but if it is LES etc then it also builds on devp2p and has certain guarantees

And in theory we could swap Ethereum out for something else, if we just like at "basic transfer of value" as a message, ~ ish (in this case it has to be committed and have gas etc)

Reply:
That's a good point to justify why we chose devp2p and Whisper which belong to Ethereum stack.